### PR TITLE
Add custom JWT keys to finance-portal service

### PR DESCRIPTION
**Pull request by Kalaston**
Some implementations may want to use JWT keys for their solutions. This feature enables them to add  private JWT via values for the service.

### CODE REVIEW

File: finance-portal/templates/secrets.yaml [hunk 1]
```diff
@@ -21,3 +21,18 @@ data:
...
+  labels:
+    app.kubernetes.io/name: {{ include "finance-portal.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+data:
+  first_key: {{ required "The JWS key is required to validate the JWS header" .Values.creds.first_key | b64enc }}
+  second_key: {{ required "The JWS key is required to validate the JWS header" .Values.creds.second_key | b64enc }}
+  third_key: {{ required "The JWS key is required to validate the JWS header" .Values.creds.third_key | b64enc }}
```
Comment (msk-) [inline finance-portal/templates/secrets.yaml]: I think probably all of this should be wrapped in `{{- if .Values.JWSKeys }} .. content .. {{- end }}`

File: finance-portal/values.yaml [hunk 2]
```diff
@@ -13,6 +13,57 @@ db:
     password: 'oyMxgZChuu'
     schema: 'central_ledger'
 
+creds:
```
Comment (msk-) [inline finance-portal/values.yaml]: Should probably be a bit more descriptive than `creds`

File: finance-portal/templates/secrets.yaml (hunk 1 above)
Comment (msk-) [inline finance-portal/templates/secrets.yaml]: Should probably use a range, something like:
```
{{- range $k, $v =: .Values.JWSKeys }}
  {{ $k }}: {{ required "blah blah" $v }}
{{- end }}
```
Also, are you sure you need to `b64enc` these? I suspect k8s does that for you.

File: finance-portal/templates/deployment.yaml [hunk 3]
```diff
@@ -73,5 +77,19 @@ spec:
...
+            items:
+              - key: first_key
+                path: first_key.key
+                mode: 511
+              - key: second_key
+                path: second_key.key
+                mode: 511
+              - key: third_key
+                path: third_key.key
+                mode: 511
```
Comment (msk-) [inline finance-portal/templates/deployment.yaml]: Wrap in a conditional block `{{- if .Values.JWSKeys }} .. content .. {{- end }}`

File: finance-portal/templates/deployment.yaml [hunk 4]
```diff
@@ -52,6 +52,10 @@ spec:
             name: {{ template "finance-portal.fullname" . }}
         - secretRef:
             name: {{ template "finance-portal.fullname" . }}
+        volumeMounts:
+        - name: {{ template "finance-portal.fullname" . }}-secrets-volume
+          mountPath: /opt/finance-portal-backend-service/secrets
+          readOnly: true
```
Comment (msk-) [inline finance-portal/templates/deployment.yaml]: Wrap in a conditional block `{{- if .Values.JWSKeys }} .. content .. {{- end }}`

File: finance-portal/templates/secrets.yaml (hunk 1 above)
Comment (Kalaston) [reply finance-portal/templates/secrets.yaml]: `b64enc` is necessary for the secret to be installed into the k8s cluster. if it's not added, helm will output different types of error.

File: finance-portal/templates/deployment.yaml [hunk 5]
```diff
@@ -73,5 +79,11 @@ spec:
             name: {{ template "finance-portal.fullname" . }}
         - secretRef:
             name: {{ template "finance-portal.fullname" . }}
+      {{- if .Values.JWTKeys }}
+      volumes:
+        - name: {{ template "finance-portal.fullname" . }}-secrets-volume
+          secret:
+            secretName: {{ template "finance-portal.fullname" . }}-certs
```
Comment (msk-) [inline finance-portal/templates/deployment.yaml]:
```suggestion
            secretName: {{ template "finance-portal.fullname" . }}-certs
            defaultMode: 511
```

File: finance-portal/values.yaml [hunk 6]
```diff
@@ -13,6 +13,58 @@ db:
     password: 'oyMxgZChuu'
     schema: 'central_ledger'
 
+JWTKeys:
+  # These are example keys, please get valid keys for your deployment
+  # The name you assign to each variable will be used as the file name.
```
Comment (msk-) [inline finance-portal/values.yaml]:
```suggestion
  # The name you assign to each variable will be used as the file name. Using the following examples, three key files will be created:
  # - /opt/finance-portal-backend-service/secrets/first_key.key
  # - /opt/finance-portal-backend-service/secrets/second_key.key
  # - /opt/finance-portal-backend-service/secrets/third_key.key
```

**Pull request merged**